core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -398,6 +398,13 @@ public YesNoMaybe supportsDynamicLoad() {
     }
 
     /**
+     * Returns true if this plugin overrides core components.
+     */
+    public boolean isCoreComponentOverride() throws IOException {
+        return "true".equals(manifest.getMainAttributes().getValue("Core-Component"));
```

Comment:
Third copy of this, maybe it could be moved into a `static` method in `jenkins.bootstrap` or something?